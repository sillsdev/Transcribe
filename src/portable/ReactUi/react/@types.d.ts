declare module 'react-avatar';
declare module 'react-contextmenu';
declare module 'react-localization';
declare module 'react-player';
declare module 'react-string-replace';
declare module 'react-toggle-switch';
declare module 'react-tooltip';
declare module 'react-thunk';