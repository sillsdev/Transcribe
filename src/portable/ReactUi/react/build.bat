call npm run build
cd ..\..\..\UpdateSharedResources
dotnet run ..\portable\ReactUi\react\build ..\portable\ReactShared\ReactShared.csproj
cd ..\portable\ReactShared
rmdir /s/q data
mkdir data
copy ..\ReactUi\data\*.xsd data
copy ..\ReactUi\data\*.xml data
copy ..\ReactUi\data\*.eaf data
cd data
mkdir localization
copy ..\..\ReactUi\data\localization\*.json localization
cd ..
rmdir /s/q react
mkdir react
cd react
mkdir build
xcopy /s/y ..\..\ReactUi\react\build\* build
cd ..
dotnet build
cd ..\ReactUi\react
call npm run jest -- --json --outputFile=.\TestResult.json
find "numFailedTests"":0," <./TestResult.json >nul
