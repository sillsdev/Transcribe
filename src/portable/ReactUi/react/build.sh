#/bin/bash
npm run build
cd ../../../UpdateSharedResources
dotnet-sdk.dotnet run ../portable/ReactUi/react/build ../portable/ReactShared/ReactShared.csproj
cd ../portable/ReactShared
rm -rf data
mkdir data
cp -a ../ReactUi/data/*.xsd data
cp -a ../ReactUi/data/*.xml data
cp -a ../ReactUi/data/*.eaf data
cd data
mkdir localization
cp -a ../../ReactUi/data/localization/*.json localization
cd ..
rm -rf react
mkdir react
cd react
mkdir build
cp -a ../../ReactUi/react/build/* build
cd ..
dotnet-sdk.dotnet build
cd ../ReactUi/react
npm run jest -- --json --outputFile=./TestResult.json
#find "numFailedTests"":0," <./TestResult.json >nul
