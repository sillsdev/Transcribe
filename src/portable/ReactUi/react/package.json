{
  "name": "transcriber",
  "version": "0.3.8",
  "private": true,
  "description": "UI for Transcriber using create-react-app ejected for sass",
  "keywords": [
    "transcribe",
    "react",
    "redux",
    "typescript",
    "webpack 3"
  ],
  "dependencies": {
    "axios": "^0.18.0",
    "promise": "8.0.1",
    "rc-slider": "^8.6.4",
    "react": "^16.6.3",
    "react-avatar": "^3.5.0",
    "react-avatar-editor": "^11.0.4",
    "react-bootstrap": "^0.32.4",
    "react-contextmenu": "^2.10.0",
    "react-dev-utils": "^5.0.2",
    "react-dom": "^16.6.3",
    "react-dropzone": "^8.0.2",
    "react-hotkeys": "^1.1.4",
    "react-icons": "^3.2.2",
    "react-localization": "^1.0.13",
    "react-player": "^1.7.1",
    "react-redux": "^6.0.0",
    "react-router-dom": "^4.3.1",
    "react-string-replace": "^0.4.1",
    "react-thunk": "^1.0.0",
    "react-toggle-switch": "^3.0.4",
    "react-tooltip": "^3.9.0",
    "redux": "^4.0.1",
    "redux-thunk": "^2.3.0",
    "reselect": "^4.0.0"
  },
  "scripts": {
    "clean": "rimraf build coverage",
    "prestart": "IF NOT EXIST node_modules\\webpack-dev-server npm i -D webpack-dev-server@2.11.3",
    "start": "node scripts/start.js",
    "build": "node scripts/build.js",
    "test": "node scripts/test.js --env=jsdom",
    "lint": "tslint \"src/**/*.tsx\"",
    "jest": "jest --env=jsdom"
  },
  "devDependencies": {
    "@types/jest": "^23.3.10",
    "@types/node": "^10.12.15",
    "@types/rc-slider": "^8.6.0",
    "@types/react": "^16.7.17",
    "@types/react-avatar-editor": "^10.3.2",
    "@types/react-bootstrap": "^0.32.15",
    "@types/react-dom": "^16.0.11",
    "@types/react-dropzone": "^4.2.2",
    "@types/react-icons": "^2.2.6",
    "@types/react-redux": "^6.0.11",
    "@types/react-router-dom": "^4.3.1",
    "autoprefixer": "7.1.6",
    "babel-jest": "20.0.3",
    "babel-loader": "7.1.2",
    "babel-preset-react-app": "^3.1.2",
    "case-sensitive-paths-webpack-plugin": "2.1.1",
    "chalk": "1.1.3",
    "css-loader": "0.28.7",
    "dotenv": "4.0.0",
    "dotenv-expand": "4.2.0",
    "enzyme": "^3.8.0",
    "enzyme-adapter-react-16": "^1.7.1",
    "extract-text-webpack-plugin": "3.0.2",
    "file-loader": "1.1.5",
    "fork-ts-checker-webpack-plugin": "^0.2.8",
    "fs-extra": "3.0.1",
    "html-webpack-plugin": "2.29.0",
    "jest": "20.0.4",
    "node-sass": "^4.11.0",
    "object-assign": "4.1.1",
    "postcss-flexbugs-fixes": "3.2.0",
    "postcss-loader": "2.0.8",
    "raf": "3.4.0",
    "resolve": "1.6.0",
    "rimraf": "^2.6.3",
    "sass-loader": "^7.1.0",
    "sinon": "^7.2.2",
    "source-map-loader": "^0.2.1",
    "style-loader": "0.19.0",
    "sw-precache-webpack-plugin": "0.11.4",
    "ts-jest": "22.0.1",
    "ts-loader": "^2.3.7",
    "tsconfig-paths-webpack-plugin": "^2.0.0",
    "tslint": "^5.7.0",
    "tslint-config-prettier": "^1.10.0",
    "tslint-react": "^3.2.0",
    "typescript": "^3.2.2",
    "uglifyjs-webpack-plugin": "1.2.5",
    "url-loader": "0.6.2",
    "webpack": "3.8.1",
    "webpack-manifest-plugin": "1.3.2",
    "whatwg-fetch": "2.0.3"
  },
  "jest": {
    "collectCoverageFrom": [
      "src/**/*.{js,jsx,ts,tsx}",
      "!**/*.d.ts"
    ],
    "setupFiles": [
      "<rootDir>/config/polyfills.js"
    ],
    "testMatch": [
      "<rootDir>/src/**/__tests__/**/*.(j|t)s?(x)",
      "<rootDir>/src/**/?(*.)(spec|test).(j|t)s?(x)"
    ],
    "testEnvironment": "node",
    "testURL": "http://localhost",
    "transform": {
      "^.+\\.(js|jsx|mjs)$": "<rootDir>/node_modules/babel-jest",
      "^.+\\.tsx?$": "<rootDir>/config/jest/typescriptTransform.js",
      "^.+\\.css$": "<rootDir>/config/jest/cssTransform.js",
      "^(?!.*\\.(js|jsx|mjs|css|json)$)": "<rootDir>/config/jest/fileTransform.js"
    },
    "transformIgnorePatterns": [
      "[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs|ts|tsx)$"
    ],
    "moduleNameMapper": {
      "^react-native$": "react-native-web"
    },
    "moduleFileExtensions": [
      "web.ts",
      "ts",
      "web.tsx",
      "tsx",
      "web.js",
      "js",
      "web.jsx",
      "jsx",
      "json",
      "node",
      "mjs"
    ],
    "globals": {
      "ts-jest": {
        "tsConfigFile": "tsconfig.test.json"
      }
    }
  },
  "babel": {
    "presets": [
      "react-app"
    ]
  },
  "eslintConfig": {
    "extends": "react-app"
  }
}
