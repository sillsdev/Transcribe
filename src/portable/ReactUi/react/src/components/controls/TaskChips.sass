.TaskChips
    display: flex
    flex-direction: row

    .ChipsBorder
        border: 1px solid #657B83
        box-sizing: border-box
        border-radius: 8px
        padding: 0 10px

    .ChipsLabel
        font-family: Roboto
        font-style: normal
        font-weight: normal
        line-height: 16px
        font-size: 12px
        text-align: center
        letter-spacing: 0.4px
        color: #657B83
