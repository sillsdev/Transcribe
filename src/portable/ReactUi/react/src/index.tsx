import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import AddManyTasks from './components/AddManyTasks';
import AvatarEdit from './components/AvatarEdit';
import MainLayout from './components/MainLayout';
import NewOrBrowseParatextProjects from './components/NewOrBrowseParatextProject';
import ProjectSettings from './components/ProjectSettings';
import SearchParatextProjects from './components/SearchParatextProjects';
import SelectProject from './components/SelectProject';
import TaskDetails from './components/TaskDetails';
import UiLang from './components/UiLang';
import UserDetails from './components/UserDetails';
import UserLogin from './components/UserLogin';
import UserSettings from './components/UserSettings';
import DevTools from './DevTool'
import './index.sass';
import registerServiceWorker from './registerServiceWorker';
import store from './store';

// https://reacttraining.com/react-router/web/example/basic
ReactDOM.render(
  <Provider store={store}>
    <Router>
      <div>
        <Route exact={true} path="/" component={UserLogin}/>
        <Route path="/project" component={SelectProject} />
        <Route path="/main" component={MainLayout} />
        <Route path="/main/avatar/Project" component={AvatarEdit} />
        <Route path="/settings" component={UserSettings}/>
        <Route path="/settings/avatar/User" component={AvatarEdit} />
        <Route path="/uilang" component={UiLang} />
        <Route path="/SearchParatextProjects" component={SearchParatextProjects} />
        <Route path="/NewOrBrowseParatextProjects" component={NewOrBrowseParatextProjects} />
        <Route path="/ProjectSettings" component={ProjectSettings} />
        <Route path="/ProjectSettings/Task" component={TaskDetails} />
        <Route path="/ProjectSettings/NewTask" component={TaskDetails} />
        <Route path="/ProjectSettings/User" component={UserDetails} />
        <Route path="/ProjectSettings/NewUser" component={UserDetails} />
        <Route path="/ProjectSettings/User/avatar/PopupUser" component={AvatarEdit} />
        <Route path="/ProjectSettings/NewUser/avatar/PopupUser" component={AvatarEdit} />
        <Route path="/ProjectSettings/AddManyTasks" component={AddManyTasks} />
        <Route path="/" component={DevTools} />
      </div>
    </Router>
  </Provider>,
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();
