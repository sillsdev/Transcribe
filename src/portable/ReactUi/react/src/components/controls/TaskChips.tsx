import * as React from 'react';
import './TaskChips.sass';

interface IProps {
    text: string[];
};

class TaskChips extends React.Component<IProps, object> {
    public render() {
        const { text } = this.props;
        const stateMapper = {
            "alltasks" :  "All Tasks",
            "complete" :  "Synced",
            "mytasks" :  "My Tasks",
            "review" :  "Transcribed",
            "transcribe" :  "In Progress",
            "upload" :  "Reviewed",
        }
        const chipsList = text.map((t: string) => (
            <div key={t} className="ChipsBorder">
                <div className="ChipsLabel">
                    <label>{stateMapper[t]}</label>
                </div>
            </div>));
        return (
            <div className="TaskChips">
                {chipsList}
            </div>

        )
    }
};

export default TaskChips;
