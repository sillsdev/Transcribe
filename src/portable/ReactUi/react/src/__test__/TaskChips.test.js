/* jshint esversion: 6 */
import * as React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import TaskChips from '../components/controls/TaskChips';

configure({ adapter: new Adapter() })

// Snapshot for TaskChips
describe('>>>Control: TaskChips --- Snapshot',()=>{
    const minProps = {
        text: ["upload", "alltasks"],
    };

    it('+++capturing Snapshot of TaskChips', () => {
        const renderedValue =  renderer.create(<TaskChips {...minProps}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: TaskChips', () => {
    let wrapper;
    const minProps = {
        text: ["upload", "alltasks"],
    };

    beforeEach(()=>{
        wrapper = shallow(<TaskChips {...minProps} />)
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
