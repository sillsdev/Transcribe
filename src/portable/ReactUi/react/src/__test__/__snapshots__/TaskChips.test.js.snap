// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: TaskChips --- Snapshot +++capturing Snapshot of TaskChips 1`] = `
<div
  className="TaskChips"
>
  <div
    className="ChipsBorder"
  >
    <div
      className="ChipsLabel"
    >
      <label>
        Reviewed
      </label>
    </div>
  </div>
  <div
    className="ChipsBorder"
  >
    <div
      className="ChipsLabel"
    >
      <label>
        All Tasks
      </label>
    </div>
  </div>
</div>
`;
